node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java

```diff
@@ -251,10 +251,8 @@ private boolean failActive(FailingNode failing) {
                 }
             }
 
-            // A parent with children gets wantToFail to avoid getting more nodes allocated to it.
-            wantToFail(failing.node(), true, lock);
-
             if (activeChildrenToFail.isEmpty()) {
+                wantToFail(failing.node(), true, lock);
```

Comment:
Won't this allow allocation of new nodes on a host we're trying to fail? Maybe we'll just have to live with that.

Reply:
Yes, I think we will have to live with that.  We could invent some other marker to prevent allocation and ensure that stays in sync with the intention of the NodeFailer, but that's a bigger fix.